core/view/src/parser/__tests__/parser.test.ts

```diff
@@ -1,7 +1,38 @@
+import { BindingParser } from '@player-ui/binding';
+import { LocalModel, withParser } from '@player-ui/data';
+import { SchemaController } from '@player-ui/schema';
 import { NodeType, Parser } from '../index';
+import { SwitchPlugin, ApplicabilityPlugin, TemplatePlugin } from '../..';
+import type { Options } from '../../plugins/options';
+import { ExpressionEvaluator } from '../../../../expressions/src';
+import type { DataModelWithParser } from '../../../../player/src';
 
+const parseBinding = new BindingParser().parse;
 describe('generates the correct AST', () => {
-  const parser = new Parser();
+  let model: DataModelWithParser;
+  let expressionEvaluator: ExpressionEvaluator;
+  let options: Options;
+  let parser: Parser;
+
+  beforeEach(() => {
+    model = withParser(new LocalModel(), parseBinding);
+    expressionEvaluator = new ExpressionEvaluator({
+      model,
+    });
+    parser = new Parser();
+    options = {
+      evaluate: expressionEvaluator.evaluate,
+      schema: new SchemaController(),
+      data: {
+        format: (binding, val) => val,
+        formatValue: (val) => val,
+        model,
+      },
+    };
+    new TemplatePlugin(options).applyParserHooks(parser);
+    new ApplicabilityPlugin().applyParser(parser);
+    new SwitchPlugin(options).applyParser(parser);
```

Comment:
Why is `TemplatePlugin` the only one that has a different API? `applyParserHooks` vs. `applyParser`